types/underscore/underscore-tests.ts

```diff
@@ -662,6 +663,139 @@ interface ChainTypeExtractor {
 
 declare const extractChainTypes: ChainTypeExtractor;
 
+// Types
+
+// Iteratee
+{
+    // functions
+    const listFunctionIteratee: _.Iteratee<_.List<StringRecord>, string> = (element, key, collection) => {
+        element; // $ExpectType StringRecord
+        key; // $ExpectType number
+        collection; // $ExpectType List<StringRecord>
+        return element.a;
+    };
+    listFunctionIteratee(stringRecordList[0], 0, stringRecordList);
+
+    const dictionaryFunctionIteratee: _.Iteratee<_.Dictionary<StringRecord>, string> = (element, key, collection) => {
+        element; // $ExpectType StringRecord
+        key; // $ExpectType string
+        collection; // $ExpectType Dictionary<StringRecord>
+        return element.a;
+    };
+    dictionaryFunctionIteratee(stringRecordDictionary['a'], 'a', stringRecordDictionary);
+
+    const unionCollectionItemFunctionIteratee: _.Iteratee<_.List<IntersectingProperties>, string | boolean> = (element, key, collection) => {
+        element; // $ExpectType IntersectingProperties
+        key; // $ExpectType number
+        collection; // $ExpectType List<IntersectingProperties>
+        return element.a;
+    };
+    unionCollectionItemFunctionIteratee(intersectingPropertiesList[0], 0, intersectingPropertiesList); // $ExpectType string | boolean
+
+    const unionCollectionFunctionIteratee: _.Iteratee<_.Dictionary<StringRecord> | StringRecord[], string> = (element, key, collection) => {
+        element; // $ExpectType StringRecord
+        key; // $ExpectType string | number
+        collection; // $ExpectType Dictionary<StringRecord> | StringRecord[]
+        return element.a;
+    };
+    unionCollectionFunctionIteratee(stringRecordDictionary['a'], 'a', stringRecordDictionary); // $ExpectType string
+
+    const anyFunctionIteratee: _.Iteratee<any, string> = (element, key, collection) => {
+        element; // $ExpectType any
+        key; // $ExpectType string | number
+        collection; // $ExpectType any
+        return element.a;
+    };
+    if (_.isFunction(anyFunctionIteratee)) {
```

Comment:
Why is this check here and not in the previous tests?

Reply:
Unlike with other types for which functions are excluded from matching `Partial<T>` because they end up with properties that functions don't have, functions match the propertyless `Partial<any>` type, so I need to do this to make `anyFunctionIteratee` not `CollectionIterator<...> | Partial<any>`. This also applies to the `isArray` check later in this file.